ui/decredmaterial/collapsible.go

```diff
@@ -119,167 +108,45 @@ func (c *Collapsible) Layout(gtx layout.Context, header func(C) D, content func(
 	})
 }
 
-func (c *CollapsibleWithOption) Layout(gtx layout.Context, header func(C) D, content func(C) D, footer func(C) D) layout.Dimensions {
-	c.handleEvents()
-
-	collapseIcon := c.theme.collapseIcon
-	expandIcon := c.theme.expandIcon
-	collapseIcon.Scale, expandIcon.Scale = 0.25, 0.25
+func (c *CollapsibleWithOption) Layout(gtx layout.Context, header, content func(C) D, more func(C)) layout.Dimensions {
+	for c.button.Clicked() {
+		c.isExpanded = !c.isExpanded
+	}
 
-	icon := expandIcon
-	if c.Collapsible.IsExpanded {
-		icon = collapseIcon
+	icon := c.collapsedIcon
+	if c.isExpanded {
+		icon = c.expandedIcon
 	}
```

Comment:
This block of code is a duplicate on Collapsible widget. It can just be a handle method with the struct type passed to it. Some other parts of the code can also be separated as functions and reused.